regular_expressions_basics/regular_expressions_basics.md

```diff
@@ -0,0 +1,338 @@
+* The space after the percent can be added as is: ` `
+* The opening parenthesis is a metacharacter, so it **must** be added to the pattern with a backslash: `\(`
+* The numbers `20` should be added as is: `20`
+* The closing parenthesis is a metacharacter, so it **must** be added to the pattern with a backslash: `\)`
+* The space after the closing parenthesis can be added as-is:  ` `
+* The letters `claim` can be added just as they appear: `claim`
+* `:` is not a metacharacter for regex, so you can add it as is, with `:`, or, if you weren't sure if `:` was a metacharacter, you can also "escape" it, giving `\:`.
+
+This means that either `25\% \(20\) claim\:` or `25% \(20\) claim:` are regular expressions that match the given text.  On the other hand, `25% (20) claim:` and `25\% (20) claim\:` do not match the given text, because they fail to escape the parentheses.
+
+</div>
+***
+
+### Tokens: Special Groups
+
+Often, we want to indicate that a group of characters are all equally valid in a location.  For example, a pattern might include any digit at all in a particular location, or any of the letters "A", "Q", or "Y" at another location.  This is very typical, and this is where regular "find" or "find and replace" tools in software fail, while regular expressions prove useful.
+
+There are several ways to indicate groups of possible characters, and we'll go over the most common ones below.
+
+Metasequences
+------  
+
+Digits and some other groups are so common they are predefined with special **metasequences**.  
+
+* Any single digit (0-9) is represented `\d`.  
+* Any whitespace character like space or tab? That's `\s`.
+* Any character at all (a wildcard)? `.`
+
+You can combine these metasquences with what we've already learned.  For example, if the pattern we're trying to define includes a single digit, the letter Z, some other character that could be anything, and the letter J, we could put that in a regular expression as `\dZ.J`.
+
+Custom Character Sets
+------
+
+Other groups of characters can be defined by square brackets.  You can type each member of the character set between square brackets.
+
+* One of the letters `Y` or `N` could be written as `[YN]` (or `[NY]`)
+* Either the number 0 or the number 1? `[01]` (or `[10]`)
+
+As we've done before, you can combine these.  For example, if the pattern we're describing begins with a single digit, then a hyphen, then either a "Y" or "N", we would write that in regex as `\d\-[YN]`.
+
+Ranges
+-----
+
+You can also use square brackets to define a range of [Unicode characters](https://en.wikipedia.org/wiki/List_of_Unicode_characters), by giving the first character in the set, a hyphen, and the last character in the set.  For example, the lowercase letters appear in alphabetical order in Unicode, so we could indicate "the letter `c`, the letter `h`, and all the letters in between" by writing `[c-h]`.
```

Comment:
Talking about a "range of unicode characters" seems a little jargon-heavy to learners who might not be familiar with unicode. I suggest rephrasing ranges to focus on numeric ranges, and then breaking this whole page into 3 sub-pages:

Character Sets:
- Metasequences (predefined sets)
- Custom Sets
Ranges:
focus on numbers, then introduce Unicode as a way to think of letters as also having orders/ranges.
Include a "learn more" box with some background on Unicode ranges actually being sequential ranges.
Letters:
All the cool alphabet stuff!
Also add an example like [A-Da-d] because maybe you don't want to use the whole alphabet.

Reply:
oooh yes I think you're right.  Will address in a new commit.

Reply:
I love the way you have divided this! 

Would you be up for making the 4 subsections actual pages rather than subsections on one long page? That will make them easier for learners to refer back to, or find exactly the part they want in the table of contents.

Reply:
@paytonk I think getting these subsections into the table of contents as subpages is they only suggestion I made that you didn't respond to.

"Not gonna do that" is a fine response, I just want to make you saw the suggestion.